shared/packages/tsr-bridge/src/sideload/Hyperdeck.ts

```diff
@@ -0,0 +1,52 @@
+import { DeviceOptionsHyperdeck } from 'timeline-state-resolver'
+import { ResourceAny, ResourceType, HyperdeckPlay, HyperdeckRecord, HyperdeckPreview } from '@shared/models'
+import { SideLoadDevice } from './sideload'
+import { LoggerLike } from '@shared/api'
+
+export class HyperdeckSideload implements SideLoadDevice {
+	constructor(private deviceId: string, _deviceOptions: DeviceOptionsHyperdeck, _log: LoggerLike) {}
+	refreshResources() {
+		return this._refreshResources()
+	}
+	async close() {
+		// Nothing to cleanup.
+	}
+	private async _refreshResources() {
```

Comment:
Is it possible to retrive a list of media files from the Hyperdeck here?
If so, this method should generate resources based on such a list.

Reply:
Yes, and it has now been implemented. However, it can't auto-populate the durations of the clips because `hyperdeck-connection` does not yet expose that information, even though it is available from the HyperDeck protocol.

See:
- https://github.com/nrkno/sofie-hyperdeck-connection/blob/master/src/commands/diskList.ts#L6-L14
- ![image](https://user-images.githubusercontent.com/873012/190428130-8b206030-2c80-4716-8e40-45f5986235aa.png)